woocommerce/woocommerce
# Marketing page - Implement "Learn about marketing a store" card

### All Submissions:


### Changes proposed in this Pull Request:

Closes https://github.com/woocommerce/woocommerce/issues/34108.

In this PR, we implement the "Learn about marketing a store" card for the new Multichannel Marketing page.

Most of the logic in the `LearnMarketing` component is inspired by the existing [`KnowledgeBase`](https://github.com/woocommerce/woocommerce/blob/58ff656764774c56bfd750752e1f3f92c74925a3/plugins/woocommerce-admin/client/marketing/components/knowledge-base/index.js) component, i.e. it has loading indicator, error handling, displaying message for zero blog posts, and displaying blog posts. The difference is mainly in the UI and the ability to expand/collapse the card.

📹 Demo video with my voice:

https://user-images.githubusercontent.com/417342/183981361-2f4b4890-b5ba-4d88-bf48-69e8574c9bd3.mov

Card UI when loading is in progress:

<img width="716" alt="image" src="https://user-images.githubusercontent.com/417342/183977292-ec2b3ef6-a64f-4e23-bbc8-7a8fe0d42d87.png">

Card UI when there is an error (the message is the same as the one in the [existing KnowledgeBase card](https://github.com/woocommerce/woocommerce/blob/ecec9eaa76cb7b2e36f79175837b71f4f64996b1/plugins/woocommerce-admin/client/marketing/components/knowledge-base/index.js#L128-L142)):

<img width="719" alt="image" src="https://user-images.githubusercontent.com/417342/183978096-898393ee-659e-4647-9b5f-292c1e10322b.png">

Card UI when there are no blog posts (the message is the same as the one in the [existing KnowledgeBase card](https://github.com/woocommerce/woocommerce/blob/ecec9eaa76cb7b2e36f79175837b71f4f64996b1/plugins/woocommerce-admin/client/marketing/components/knowledge-base/index.js#L115-L126)):

<img width="705" alt="image" src="https://user-images.githubusercontent.com/417342/183978290-e865f401-8a5b-4ca3-9a8e-031b28d6688a.png">

Card UI when blog posts are retrieved and displayed successfully:

<img width="712" alt="image" src="https://user-images.githubusercontent.com/417342/183978466-2ca9698a-7826-4e75-8eb6-48a7af1ab8c8.png">

Other technical notes:

- I move the existing `ReadBlogMessage` component and use it in this new `LearnMarketing` component.

### How to test the changes in this Pull Request:

1. Enable the multichannel marketing feature in `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`.
2. Visit `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`.
3. The "Learn about marketing a store" card should be collapsed by default.
4. Expand the card.
5. When blog posts is loading, there should be a loading indicator.
6. When there is an error in retrieving the blog posts, there should be an error message.
7. When there are no blog posts, there should be a message indicating "No posts yet".
8. When blog posts are loaded successfully, they should be displayed in the card, with two posts in one page. You should be able to page through them by clicking on the pagination component in the card footer.

### Other information:


### FOR PR REVIEWER ONLY:
